adds some description for adding dist extensions (#302)
opencontainers/distribution-spec: README.md

The [GitHub milestones](https://github.com/opencontainers/distribution-spec/milestones) lay out the path to the future improvements.

# Extensions

> The Distribution Specification Project is includes a process and API for prototyping

Reviewer: ```suggestion
The Distribution Specification Project includes a process and API for prototyping
```

> The Distribution Specification Project is including, herein, a process and API for prototyping

Reviewer: ```suggestion
The Distribution Specification Project is includes a process and API for prototyping
```

and testing extensions to the Distribution API.

We invite contributions, comments, and reviews to these extensions.

> These extensions will only move forward with significant support from registies, registry clients, and users.

Reviewer: ```suggestion
These extensions will only move forward with significant support from registries, registry clients, and users.
```

The [GitHub milestones](https://github.com/opencontainers/distribution-spec/milestones) lay out the path to the future improvements.

# Extensions

The Distribution Specification Project is including, herein, a process and API for prototyping
and testing extensions to the Distribution API.

We invite contributions, comments, and reviews to these extensions.
These extensions will only move forward with significant support from registies, registry clients, and users.

Disclaimer: Unless stated otherwise, extension prototypes are subject to change. Extensions
should NOT be considered as final or approved official extensions to the OCI Distribution API until they

> have been included in an OCI Distribution Specification.

Reviewer: It seems like this is the case now. Should we close this PR @mikebrow ?

Author: I need to update this PR to point to https://github.com/opencontainers/distribution-spec/blob/main/extensions/README.md

> Disclaimer: Unless stated otherwise, extension prototypes are subject to change. Extensions
> should NOT be considered as final or approved official extensions to the OCI Distribution API until they
> have been included in an OCI Distribution Specification.

Reviewer: ```suggestion
Please see [here](https://github.com/opencontainers/distribution-spec/blob/main/extensions/README.md) for more details.
```

> 
> ## Extensions API - (status: WIP)
> 
> The Distribution maintainers recognize that an API may be necessary, or at least potentially advantageous, for specifying
> extensions to the Distribution API. Thus, we propose to add an [Extensions API for Distribution](EXTENSIONS.md).
> 

Reviewer: ```suggestion
```
